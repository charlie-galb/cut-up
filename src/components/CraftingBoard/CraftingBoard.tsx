import React from 'react'

import { chunkContainer } from "../../types/chunkContainer"

import { PasteBoard } from "../PasteBoard/PasteBoard"
import { WorkInProgress } from "../WorkInProgress/WorkInProgress"

interface Props {
    chunkContainers: {
      [key: string]: chunkContainer
    }
    lineOrder: string[]
    addLine: (arg: void) => void
    wordChunks: {
        [key: string]: string
      }
}

export const CraftingBoard = (props: Props) => {
    return (
        <div className="crafting-container">
            <PasteBoard wordChunks={props.wordChunks} chunkContainer={props.chunkContainers['chunk-container-1']} />
            <WorkInProgress wordChunks={props.wordChunks} lineOrder={props.lineOrder} chunkContainers={props.chunkContainers} addLine={props.addLine}/>
        </div>
    )
}
