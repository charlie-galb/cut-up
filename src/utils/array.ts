import { arrayMove as dndKitArrayMove } from "@dnd-kit/sortable";

export const removeAtIndex = (array: string[], index: number) => {
  return [...array.slice(0, index), ...array.slice(index + 1)];
};

export const insertAtIndex = (array: string[], index: number, item: string) => {
  return [...array.slice(0, index), item, ...array.slice(index)];
};

export const arrayMove = (array: string[], oldIndex: number, newIndex: number) => {
  return dndKitArrayMove(array, oldIndex, newIndex);
};