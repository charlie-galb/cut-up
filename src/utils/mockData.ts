import { chunkContainer } from "../types/chunkContainer"

export const mockChunks = {
    "1": "one",
    "2": "two"
}

export const mockChunkIDs = ["1", "2"]

export const mockContainer1: chunkContainer = {
    id: 'chunk-container-1',
    title: 'pasteboard',
    nestedChunkIDs: mockChunkIDs
  }

export const mockContainer2: chunkContainer = {
id: 'chunk-container-2',
title: 'Line-1',
nestedChunkIDs: mockChunkIDs
}

export const mockContainers = {
    [mockContainer1.id]: mockContainer1,
    [mockContainer2.id]: mockContainer2,
}

export const mockLineOrder = [mockContainer1.id, mockContainer2.id]