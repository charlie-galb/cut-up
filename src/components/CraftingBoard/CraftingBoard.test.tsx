import React from 'react'
import { render } from '@testing-library/react'

import { CraftingBoard } from "./CraftingBoard"

import { mockContainers, mockLineOrder, mockChunks } from "../../utils/mockData"

describe("CraftingBoard", () => {
    it('renders correctly', () => {
        const board = render(
            <CraftingBoard
                chunkContainers={mockContainers}
                lineOrder={mockLineOrder}
                wordChunks={mockChunks}
                activeId={""}
                poemAsText={""}
                setChunkContainers={jest.fn()}
                setLineOrder={jest.fn()}
                setPoemAsText={jest.fn()}
                />
        )
        expect(board).toMatchSnapshot()
    })
})