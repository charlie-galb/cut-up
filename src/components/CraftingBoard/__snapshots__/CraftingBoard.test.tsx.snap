// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CraftingBoard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="crafting-background"
      >
        <div
          class="crafting-container"
        >
          <div
            class="pasteboard-container"
          >
            <div>
              <div
                class="unused-snippets"
                data-testid="pasteboard"
              >
                <div>
                  <button
                    aria-describedby=""
                    aria-roledescription="sortable"
                    class="text-snippet"
                    role="button"
                    tabindex="0"
                    text="one"
                  >
                    one
                  </button>
                </div>
                <div>
                  <button
                    aria-describedby=""
                    aria-roledescription="sortable"
                    class="text-snippet"
                    role="button"
                    tabindex="0"
                    text="two"
                  >
                    two
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="wip-container"
          >
            <div
              class="lines-container"
            >
              <div>
                <div
                  class="line"
                  data-testid="pasteboard"
                >
                  <div>
                    <button
                      aria-describedby=""
                      aria-roledescription="sortable"
                      class="text-snippet"
                      role="button"
                      tabindex="0"
                      text="one"
                    >
                      one
                    </button>
                  </div>
                  <div>
                    <button
                      aria-describedby=""
                      aria-roledescription="sortable"
                      class="text-snippet"
                      role="button"
                      tabindex="0"
                      text="two"
                    >
                      two
                    </button>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="line"
                  data-testid="Line-1"
                >
                  <div>
                    <button
                      aria-describedby=""
                      aria-roledescription="sortable"
                      class="text-snippet"
                      role="button"
                      tabindex="0"
                      text="one"
                    >
                      one
                    </button>
                  </div>
                  <div>
                    <button
                      aria-describedby=""
                      aria-roledescription="sortable"
                      class="text-snippet"
                      role="button"
                      tabindex="0"
                      text="two"
                    >
                      two
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <button
                class="btn add-line-btn"
                data-testid="add-line-btn"
              >
                +
              </button>
            </div>
            <div>
              <button
                class="btn add-line-btn"
                data-testid="add-line-btn"
              >
                -
              </button>
            </div>
          </div>
        </div>
        <div
          class="textify-button-container"
          data-testid="textify-button-container"
        >
          <button
            class="btn textify-btn"
            data-testid="textify-btn"
          >
            Textify
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="crafting-background"
    >
      <div
        class="crafting-container"
      >
        <div
          class="pasteboard-container"
        >
          <div>
            <div
              class="unused-snippets"
              data-testid="pasteboard"
            >
              <div>
                <button
                  aria-describedby=""
                  aria-roledescription="sortable"
                  class="text-snippet"
                  role="button"
                  tabindex="0"
                  text="one"
                >
                  one
                </button>
              </div>
              <div>
                <button
                  aria-describedby=""
                  aria-roledescription="sortable"
                  class="text-snippet"
                  role="button"
                  tabindex="0"
                  text="two"
                >
                  two
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="wip-container"
        >
          <div
            class="lines-container"
          >
            <div>
              <div
                class="line"
                data-testid="pasteboard"
              >
                <div>
                  <button
                    aria-describedby=""
                    aria-roledescription="sortable"
                    class="text-snippet"
                    role="button"
                    tabindex="0"
                    text="one"
                  >
                    one
                  </button>
                </div>
                <div>
                  <button
                    aria-describedby=""
                    aria-roledescription="sortable"
                    class="text-snippet"
                    role="button"
                    tabindex="0"
                    text="two"
                  >
                    two
                  </button>
                </div>
              </div>
            </div>
            <div>
              <div
                class="line"
                data-testid="Line-1"
              >
                <div>
                  <button
                    aria-describedby=""
                    aria-roledescription="sortable"
                    class="text-snippet"
                    role="button"
                    tabindex="0"
                    text="one"
                  >
                    one
                  </button>
                </div>
                <div>
                  <button
                    aria-describedby=""
                    aria-roledescription="sortable"
                    class="text-snippet"
                    role="button"
                    tabindex="0"
                    text="two"
                  >
                    two
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div>
            <button
              class="btn add-line-btn"
              data-testid="add-line-btn"
            >
              +
            </button>
          </div>
          <div>
            <button
              class="btn add-line-btn"
              data-testid="add-line-btn"
            >
              -
            </button>
          </div>
        </div>
      </div>
      <div
        class="textify-button-container"
        data-testid="textify-button-container"
      >
        <button
          class="btn textify-btn"
          data-testid="textify-btn"
        >
          Textify
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
